### PR TITLE
gnrc_sixlowpan: document submodules according to #8511

### PR DESCRIPTION
**Pull request by miri64**
### Contribution description
As a consequence of #8511 I reworked the submodules of `gnrc_sixlowpan` in [this project](https://github.com/RIOT-OS/RIOT/projects/18) but never actually documented the new behavior anywhere except in the issue. This PR fixes that.

### Testing procedure
`make doc` should produce nicely rendered output ;-).

### Issues/PRs references
Documents behavior outlined in #8511.

### CODE REVIEW

File: sys/include/net/gnrc/sixlowpan.h [hunk 1]
```diff
@@ -11,11 +11,45 @@
  * @ingroup     net_gnrc
  * @brief       GNRC's 6LoWPAN implementation
  *
- * This module is for usage with the @ref net_gnrc_netapi.
+ * # Internal API and sub-modules
+ *
+ * Internally, @ref net_gnrc_sixlowpan is sub-divided into several sub-modules.
+ * This implement certain features of the 6LoWPAN standard. Currently
```
Comment (smlng) [inline sys/include/net/gnrc/sixlowpan.h]: typo, either:

- `this implement[s]`
- `[these] implement`

or for the latter variant: `each implementing ...`

File: sys/include/net/gnrc/sixlowpan.h [hunk 2]
```diff
@@ -11,11 +11,45 @@
...
+ *
+ * Each sub-module has a `send` and `recv` function prefixed by their
+ * respective sub-module name with the following signatures
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * void send(gnrc_pktsnip_t *pkt, void *ctx, uint8_t page);
+ * void recv(gnrc_pktsnip_t *pkt, void *ctx, uint8_t page);
+ * ~~~~~~~~~~~~~~~~~~~~~
+ *
+ * A 6LoWPAN frame `pkt` is supposed to pass the sub-modules sequentially in
```
Comment (smlng) [inline sys/include/net/gnrc/sixlowpan.h]: > supposed

? sounds vague, better state explicitly how it should be implemented.

File: sys/include/net/gnrc/sixlowpan.h [hunk 3]
```diff
@@ -11,11 +11,45 @@
...
+ * the order of its dispatches on receive or the step that makes most sense next
+ * on send. After it was passed into another sub-module using the respective
+ * `send`/`recv` function a sub-module is not supposed to operate on `pkt`
```
Comment (smlng) [inline sys/include/net/gnrc/sixlowpan.h]: again, write down how it should be implemented, don't use `supposed`

File: sys/include/net/gnrc/sixlowpan.h [hunk 4]
```diff
@@ -11,11 +11,45 @@
...
+ * anymore.
+ *
+ * The `ctx` parameter can be used to provide data structures of a sub-module to
+ * the next sub-module that might need to modify them (e.g. reassembly
```
Comment (smlng) [inline sys/include/net/gnrc/sixlowpan.h]: `might` also sounds vague, to me `ctx` should only be used (passed) when the next module must/should modify data structures, hence write `... to the next sub-module if that needs to modify them (e.g. ...`

File: sys/include/net/gnrc/sixlowpan.h (hunk 2 above)
Comment (miri64) [reply sys/include/net/gnrc/sixlowpan.h]: Why is supposed vague? Yes, this documents also how it is implemented in the current implementation, but also is a guide to implement new sub-modules.

File: sys/include/net/gnrc/sixlowpan.h (hunk 3 above)
Comment (miri64) [reply sys/include/net/gnrc/sixlowpan.h]: Again, I don't know why you think `supposed` is vague... Should I use MUST?

File: sys/include/net/gnrc/sixlowpan.h (hunk 2 above)
Comment (smlng) [reply sys/include/net/gnrc/sixlowpan.h]: maybe its subjective (or bc I'm used to RFC language), but I would rather use SHOULD (or MUST) - `supposed` sounds soft(er) to me.

**Comment by smlng** [comment]
please squash

**Comment by miri64** [comment]
Squashed

**Pull request merged**